lib/pub_sub/configuration.rb

```diff
@@ -41,9 +42,10 @@ def subscribe_to(service_name, messages: [])
     # You can specify the sender, messages, and topic - all independently.
     # If topic is not specified, the value for {service_identifier} will be used.
     def subscribe_to_custom(service_identifier, messages: [], topic: nil)
-      topic ||= service_identifier
       @subscriptions[service_identifier] = messages
-      @topics[service_identifier] = topic
+      @topics[service_identifier] = topic || service_identifier
+      # Assume handler class names based on conventions, ex: deal_update -> DealUpdate
+      @handlers = Hash[*messages.map { |m| [m, m.camelize.constantize] }.flatten]
```

Comment:
@gwshaw 

I think what ashoda meant is:

There are use cases where we want to subscribe to multiple services, for ex. https://github.com/westfield/collection_service/blob/ecee3f3df62109d5893e520496ad3bd8ddc1b3ae/config/initializers/pub_sub.rb

In that case, only the last line would prevail in the `PubSub.config` singleton, because we are overwriting handlers rather than putting them together